modules/drivers/sensors/hx711/modHX711c.c

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2023 by Thorsten von Eicken.
+
+#include "xsPlatform.h"
+#include "xsmc.h"
+#include "modGPIO.h"
+#include "mc.xs.h" // for xsID_* constants
+
+#define xsmcVar(x) xsVar(x)
+
+typedef struct {
+  modGPIOConfigurationRecord clk;
+  modGPIOConfigurationRecord din;
+  int gain;
+} hx711_data;
+
+void xs_HX711_init(xsMachine *the) {
+  if (xsmcArgc != 3) xsUnknownError("invalid arguments");
+  int clk_pin = xsmcToInteger(xsArg(0));
+  int din_pin = xsmcToInteger(xsArg(1));
+
+  hx711_data *data = c_malloc(sizeof(hx711_data));
+  if (data == NULL) xsUnknownError("can't allocate data");
+  data->gain = xsmcToInteger(xsArg(2));
+  if (modGPIOInit(&data->clk, NULL, clk_pin, kModGPIOOutput))
+    xsUnknownError("can't init clk pin");
+  modGPIOWrite(&data->clk, 0);
+  if (modGPIOInit(&data->din, NULL, din_pin, kModGPIOInput))
+    xsUnknownError("can't init dat pin");
```

Comment:
If any of these throw, `data` is orphaned (even the call to `xsmcToInteger` can call in type coercion cases). Error clean-up isn't pretty, but here's what I'd probably do here:

```c
  modGPIOConfigurationRecord clk;
  modGPIOConfigurationRecord din;
  int clk_pin = xsmcToInteger(xsArg(0));
  int din_pin = xsmcToInteger(xsArg(1));
  int gain = xsmcToInteger(xsArg(2));

  if (modGPIOInit(&clk, NULL, clk_pin, kModGPIOOutput))
    xsUnknownError("can't init clk pin");
  if (modGPIOInit(&din, NULL, din_pin, kModGPIOOutput)) {
    modGPIOInit(&clk);
    xsUnknownError("can't init din pin");
  }
  hx711_data *data = c_malloc(sizeof(hx711_data));
  if (data == NULL) {
    modGPIOUninit(&clk);
    modGPIOUninit(&din);
    xsUnknownError("can't allocate data");
  }
  data->clk = clk;
  data->din = din;
  data->gain = gain;